src/ceph_mon.cc

```diff
@@ -158,25 +158,25 @@ int check_mon_data_empty()
 
 static void usage()
 {
-  cerr << "usage: ceph-mon -i monid [flags]" << std::endl;
-  cerr << "  --debug_mon n\n";
-  cerr << "        debug monitor level (e.g. 10)\n";
-  cerr << "  --mkfs\n";
-  cerr << "        build fresh monitor fs\n";
-  cerr << "  --force-sync\n";
-  cerr << "        force a sync from another mon by wiping local data (BE CAREFUL)\n";
-  cerr << "  --yes-i-really-mean-it\n";
-  cerr << "        mandatory safeguard for --force-sync\n";
-  cerr << "  --compact\n";
-  cerr << "        compact the monitor store\n";
-  cerr << "  --osdmap <filename>\n";
-  cerr << "        only used when --mkfs is provided: load the osdmap from <filename>\n";
-  cerr << "  --inject-monmap <filename>\n";
-  cerr << "        write the <filename> monmap to the local monitor store and exit\n";
-  cerr << "  --extract-monmap <filename>\n";
-  cerr << "        extract the monmap from the local monitor store and exit\n";
-  cerr << "  --mon-data <directory>\n";
-  cerr << "        where the mon store and keyring are located\n";
+  cout << "usage: ceph-mon -i monid [flags]" << std::endl;
```

Comment:
To some extent, one could argue that we should not output to `dout` or `derr` before we prefork. However, taking such a broad criteria would potentially ignore some useful messages to have in the log.

E.g., having the messages during `mkfs` being outputted to `cerr` and `cout` would be nice, as they'd be cleaner when presented to the user if called via the CLI; on the other hand, we don't get them written to the log if `ceph-mon`, and if the monitor is not invoked with `-d`, then we would really like to have them written somewhere. Granted, the monitor will only fork later on, but the caller may be ignoring any output from the daemon.

The more I think about it, the more divided I am on what's the best course of action. Any thoughts?